File: android/src/main/java/org/ligi/passandroid/json_adapter/ColorAdapter.java
package org.ligi.passandroid.json_adapter;

import android.graphics.Color;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

import org.ligi.passandroid.model.PassImpl;

public class ColorAdapter {
    @ToJson
    String toJson(@PassImpl.HexColor int rgb) {
        return String.format("#%06x", rgb);
    }

    @FromJson
    @PassImpl.HexColor
    int fromJson(String rgb) {
        return Color.parseColor(rgb);
    }
}

File: android/src/main/java/org/ligi/passandroid/json_adapter/ColorAdapter.kt
package org.ligi.passandroid.json_adapter

import android.graphics.Color
import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson
import org.ligi.passandroid.model.pass.PassImpl

class ColorAdapter {
    @ToJson
    internal fun toJson(@PassImpl.HexColor rgb: Int) = String.format("#%06x", rgb)

    @FromJson
    @PassImpl.HexColor
    internal fun fromJson(rgb: String) = Color.parseColor(rgb)

}
